lib/ansible/plugins/action/gather_facts.py

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from collections import MutableMapping
+
+from ansible import constants as C
+from ansible.plugins.action import ActionBase
+
+
+class ActionModule(ActionBase):
+
+    def run(self, tmp=None, task_vars=None):
+        ''' handler for package operations '''
+
+        self._supports_check_mode = True
+        self._supports_async = True
+
+        result = super(ActionModule, self).run(tmp, task_vars)
+        result['ansible_facts'] = {}
+
+        for fact_module in C.config.get_config_value('FACTS_MODULES', variables=task_vars):
```

Comment:
This looks like it's only configurable via the config.  We should have it replace the gather_subset functionality as well.

Reply:
Hmm... could
```
  vars:
    - name: ansible_facts_modules
```
Serve that purpose?  (And in inventory as well)?  Need to update documentation for that.  Also need to document how config works for 2.6.

Reply:
yes, its by config, but that includes vars